### PR TITLE
Implement annotated tag support.

### LINKED ISSUE #4: Annotated tag support (?)
Tags created with `-a -m <message>` are not reported in the `tag` field when the repo's `HEAD` is at the sha pointed to by the tag. :crying_cat_face: 

It appears that this is because annotated tags are full git objects with their own sha (and authors/timestamps/messages) separate from the ones they point to, so the tag's sha never matches when doing a [straight](https://github.com/rwjblue/git-repo-info/blob/master/index.js#L37) [comparison](https://github.com/rwjblue/git-repo-info/blob/master/index.js#L57).

After a fair amount of `.git` dir spelunking, it appears that there is no direct tag sha → commit sha mapping outside of the tag's binary git object. I _think_ that means this is a lost cause, but maybe you have some tricks up your sleeve?

### PR DESCRIPTION
**Pull request by rwjblue**
Closes #4.

### CODE REVIEW

**Comment by rwjblue** [comment]
Released in v1.1.0.

File: index.js [hunk 1]
```diff
@@ -43,19 +44,46 @@ function findPackedTag(gitPath, sha) {
   }
 }
 
+function commitForTag(gitPath, tag) {
+  var tagPath = path.join(gitPath, 'refs', 'tags', tag);
+  var taggedObject = fs.readFileSync(tagPath, { encoding: 'utf8' }).trim();
+  var objectPath = path.join(gitPath, 'objects', taggedObject.slice(0, 2), taggedObject.slice(2));
```
Comment (slindberg) [inline index.js:50]: Oh man.

File: index.js [hunk 2]
```diff
@@ -43,19 +44,46 @@ function findPackedTag(gitPath, sha) {
...
+
+  if (!zlib.inflateSync) {
+    // we cannot support annotated tags on node v0.10 because
+    // zlib does not allow sync access
+    return taggedObject;
+  }
+
+  var objectContents = zlib.inflateSync(fs.readFileSync(objectPath)).toString();
```
Comment (slindberg) [inline index.js:58]: OH MAN.

File: index.js [hunk 3]
```diff
@@ -43,19 +44,46 @@ function findPackedTag(gitPath, sha) {
...
+
+  // 'tag 172\u0000object c1ee41c325d54f410b133e0018c7a6b1316f6cda\ntype commit\ntag awesome-tag\ntagger Robert Jackson <robert.w.jackson@me.com> 1429100021 -0400\n\nI am making an annotated tag.\n'
+  if (objectContents.slice(0,3) === 'tag') {
+    var sections = objectContents.split(/\0|\n/);
+    var sha = sections[1].slice(7);
```
Comment (slindberg) [inline index.js:63]: _OH MAN_.

**Comment by slindberg** [comment]
Truly amazing. You are a gentleman and a hacker.

**Comment by rwjblue** [comment]
Not knowing how it worked internally was killing me.

**Comment by slindberg** [comment]
It's a slippery slope to becoming a git core contributor. What's next? BBQs at Linus' house?

**Comment by rwjblue** [comment]
I do like good BBQ, I might be a goner....

**Comment by hoppantut** [comment]
This still does not work for me. I created an annotated tag
`git tag -a 1.0.1 -m 'Hotfix'`

Checking out this tag (git checkout 1.0.1), this still gives me `null` for info.tag
```
var getRepoInfo = require('git-repo-info');

var info = getRepoInfo();

info.sha //=> Works as expected
info.tag //=> Gives null
```
Did I misunderstand how this is supposed to work?

**Pull request merged**